semantics/cpp14/language/translation/decl/enum.k

```diff
@@ -0,0 +1,55 @@
+module CPP-DECL-ENUM-SYNTAX
+     imports BOOL-SYNTAX
+     imports LIST
+     imports COMMON-SORTS
+     imports CPP-SORTS
+     imports CPP-TYPING-SYNTAX
+
+     syntax CPPSimpleType ::= getUnderlyingType(CPPSimpleEnumType) [strict, function]
+
+endmodule
+
+module CPP-DECL-ENUM
+     imports CPP-DECL-ENUM-SYNTAX
+     imports C-CONFIGURATION
+     imports K-REFLECTION
+     imports COMMON-SYNTAX
+     imports COMPAT-SYNTAX
+     imports CPP-ABSTRACT-SYNTAX
+     imports CPP-DYNAMIC-SYNTAX
+     imports CPP-ENV-SYNTAX
+     imports CPP-SYNTAX
+     imports CPP-BITSIZE-SYNTAX
+
+     syntax KItem ::= addToCurrentNamespace(CId, EnumID)
+     rule <k> addToCurrentNamespace(X::CId, enumID(I::Int)) => . ... </k>
+          <curr-scope> namespaceScope(N::Namespace) </curr-scope>
+          <curr-tu> Tu::String </curr-tu>
+          <tu-id> Tu </tu-id>
+          <ns-id> N </ns-id>
+          <ntypes> NT::Map => NT[X, Enum() <- t(noQuals, .Set, cppEnumType(enumID(I)))] </ntypes>
+
+
+     syntax Bool ::= isScoped(Scoped) [function]
+     rule isScoped(Scoped()::Scoped) => true
+     rule isScoped(Unscoped()::Scoped) => false
+
+     // second param -underlying type
+     syntax KItem ::= newEnum(EnumID, CPPType, Bool)
+     rule <k> newEnum(enumID(I:Int)::EnumID, UT::CPPType, S:Bool) => . ... </k>
+          <curr-tu> Tu::String </curr-tu>
+          <tu-id> Tu </tu-id>
+          (.Bag => <cppenum>...
+               <enum-id> I </enum-id>
+               <enum-ut> UT </enum-ut>
+               <scoped> S </scoped>
+           ...</cppenum>)
+
+
+     rule <k> EnumDef(X::CId, NoNNS(), S::Scoped, UT::CPPType, Enumerators::List) =>
```

Comment:
alright. I'm okay if we defer that to a later pull request, but if we do that can we change the matching on a variable to matching on an empty list so that the code gets stuck if it reaches a point where it needs a feature we haven't implemented yet?